File: src/Lab/Lab1.java
package Lab;

import java.util.Scanner;

public class Lab1 {
    public static int[] twoSumBruteForce(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[] {i, j};
                }
            }
        }
        throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter the number of elements in the array:");
        int n = scanner.nextInt();
        int[] nums = new int[n];

        System.out.println("Enter the elements (integers only):");
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt();
        }

        System.out.println("Enter the target sum:");
        int target = scanner.nextInt();

        try {
            int[] indices = twoSumBruteForce(nums, target);
            System.out.println("Indices of the two numbers are: " + indices[0] + " and " + indices[1]);
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        }

        scanner.close();
    }
}

File: LabPython/Lab1.py
def two_sum_brute_force(nums, target):
    for i in range(len(nums)):
        for j in range(i + 1, len(nums)):
            if nums[i] + nums[j] == target:
                return [i, j]
    raise ValueError("No two sum solution")

def main():
    n = int(input("Enter the number of elements in the array: "))
    nums = []

    print("Enter the elements (integers only):")
    for _ in range(n):
        nums.append(int(input()))

    target = int(input("Enter the target sum: "))

    try:
        indices = two_sum_brute_force(nums, target)
        print(f"Indices of the two numbers are: {indices[0]} and {indices[1]}")
    except ValueError as e:
        print(e)

if __name__ == "__main__":
    main()